c++ training/LinkedLists/LinkedLists/DoublyLinkedList.h

```diff
@@ -0,0 +1,323 @@
+	{
+		return GetNodeAt(Index).Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& GetHead()
+	{
+		return Head->Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& GetHead() const
+	{
+		return Head->Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& GetTail()
+	{
+		return Tail->Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& GetTail() const
+	{
+		return Tail->Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	bool IsEmpty() const
+	{
+		return Size == 0;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return Size;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	void AddHead(const T& Data)
+	{
+		if (IsEmpty())
+		{
+			Head = new FNode(Data);
+			Tail = Head;
+
+			Size = 1;
+			return;
+		}
+
+		Head->Previous = new FNode(Data, Head);
+		Head = Head->Previous;
+
+		Size++;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	void AddTail(const T& Data)
+	{
+		if (IsEmpty())
+		{
+			AddHead(Data);
+			return;
+		}
+
+		Tail->Next = new FNode(Data, nullptr, Tail);
+		Tail = Tail->Next;
+
+		Size++;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	void Insert(const T& Data, const int Index)
+	{
+		if (Index < 0 || Index > Size)
+		{
+			return;
+		}
+
+		if (Index == 0)
+		{
+			AddHead(Data);
+			return;
+		}
+		else
+		{
+			FNode& PreviousNode = GetNodeAt(Index - 1);
+			FNode* CurrentPtr = PreviousNode.Next;
+			PreviousNode.Next = new FNode(Data, CurrentPtr, &PreviousNode);
+
+			if (Index == Size)
+			{
+				Tail = Tail->Next;
+			}
+			else
+			{
+				CurrentPtr->Previous = PreviousNode.Next;
```

Comment:
also, you keep using .Next What would JD think about you if he see this :C  xd

Reply:
actually I think it's a good way to work

Reply:
Soo... do you approve? or would you still recommend I change it? it's kinda unclear haha